airflow/www/security.py

```diff
@@ -320,33 +317,75 @@ def get_accessible_dags(self, user_actions, user, session=None):
             for permission in role.permissions:
                 resource = permission.view_menu.name
                 action = permission.permission.name
-                if action in user_actions:
+                if action not in user_actions:
+                    continue
+
+                if resource.startswith(DAG_PREFIX):
+                    resources.add(resource[len(DAG_PREFIX) :])
+                else:
                     resources.add(resource)
 
-        if bool({'Dag', 'all_dags'}.intersection(resources)):
+        if 'Dag' in resources:
```

Comment:
I think this is my only concern -- it's not clear from the permission name that "can read on Dag" (as this would be shown in the FAB Permissions screens) would be _all_ dags.

Hmmm.

Reply:
I think this should _possibly_ be the exception to the resource naming convention, and call this "All Dags"

Reply:
I think that's a good idea. While it is an exception, it does remove some ambiguity to clarify the name.